plugins/modules/proxmox_ipam_info.py

```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2025, Jana Hoch <janahoch91@proton.me>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+module: proxmox_ipam_info
+short_description: Retrieve information about ipams
+description:
+  - Retrieve all IPs under IPAM
+  - get IP by vmid
+author: 'Jana Hoch <janahoch91@proton.me> (!UNKNOWN)'
+options:
+  vmid:
+    description:
+      - Get ip of a VM under IPAM.
+    type: int
+  ipam:
+    description:
+      - Limit results to single ipam
+    type: str
+
+extends_documentation_fragment:
+  - community.proxmox.proxmox.actiongroup_proxmox
+  - community.proxmox.proxmox.documentation
+  - community.proxmox.attributes
+  - community.proxmox.attributes.info_module
+"""
+
+EXAMPLES = r"""
+- name: Get all IPs under all IPAM
+  community.proxmox.proxmox_ipam_info:
+    api_user: "{{ pc.proxmox.api_user }}"
+    api_token_id: "{{ pc.proxmox.api_token_id }}"
+    api_token_secret: "{{ vault.proxmox.api_token_secret }}"
+    api_host: "{{ pc.proxmox.api_host }}"
+    validate_certs: no
```

Comment:
I know it's a bit nitpicky, but I think it is nice to use examples, which pass `ansible-lint`.

```suggestion
    validate_certs: false
```